### DISCUSSION TITLE
Error while starting docker

### DISCUSSION THREAD

**Discussion by patrickelectric** [lowlighter/metrics#843] [Q&A] [answered]
Command:
`docker run --volume=$PWD/settings.json:/metrics/settings.json --env INPUT_TOKEN=$TOKEN --env INPUT_USER=$MYUSER --volume=/tmp:/renders ghcr.io/lowlighter/metrics:latest`

```
Metrics                                                         │  
Setup                                                           │ complete
Version                                                         │ 3.19.0
TypeError: Cannot read properties of undefined (reading 'formats')
    at file:///metrics/source/app/action/index.mjs:131:183
::error::Cannot read properties of undefined (reading 'formats')
```

**Comment by lowlighter** [comment, marked as answer]
Hi!

Indeed, seems that defaults values are not correctly applied, this is probably a regression due to the presets feature introduction.
I'll look into it 👍

**Reply by lowlighter** [reply]
Should eventually solved by #852 once it lands

**Reply by patrickelectric** [reply]
It's working now, thanks!

**Comment by adamsimonini** [comment]
@lowlighter, the Docker instructions are unclear to me. There are two env variables being set up for the Docker container:

INPUT_USER=
&
INPUT_TOKEN=

How do I know what these values should be? I am trying to use Metrics for a repo owned by an org, so I assume INPUT_USER should be the name of the org. Is that so?

Also, what is INPUT_TOKEN?

Thank you.

**Reply by lowlighter** [reply]
Hi 👋  ! 

So the env var prefixed with `INPUT_` actually corresponds to what is also used when using GitHub Actions.
So in this case `INPUT_USER` is indeed target user/org, and `INPUT_TOKEN` should be set to a personal access token (classic).

You can find more informations about these inputs in the [`🧱 Core`](https://github.com/lowlighter/metrics/blob/master/source/plugins/core/README.md#token) and how to create a token [here](https://github.com/lowlighter/metrics/blob/master/.github/readme/partials/documentation/setup/action.md#1%EF%B8%8F-create-a-github-personal-token) (note that the new fine grained token are not supported yet, as GitHub hasn't implemented it for the GraphQL API upon which this app relies heavily)

Hope this help !

**Reply by adamsimonini** [reply]
@lowlighter, thank you for your help. I have it running now! However, I don't see the rendered svg anywherew.

I'm very confused. I don't have a persisting volume after I run the Docker image. I created a /renders, /metrics_renders, and /tmp folder at the root of my repo, but none of them contain the rendered svg. I'm excited about your project, but I sadly keep running into issues getting it to work for me.

![image](https://user-images.githubusercontent.com/20141699/212118027-fd774c43-c178-4afe-903a-84360c9505aa.png)

**Reply by lowlighter** [reply]
Yeah using directly the docker image isn't really the most intuitive way to use this app as it wasn't really intended to be offered as a way to consume it, I might check to tweak this to make it more user friendly in the future 😅 

___

If you still want to try using it as a docker container, you may be able to run it by checking how the GitHub Action and the workflows do in on this repo.

So the GitHub Action is actually consuming the docker image this way:
https://github.com/lowlighter/metrics/blob/4141dd5a5bff259430398d328027afeeb74b03d8/action.yml#L1491-L1492

So in the above command, `$METRICS_RENDERS` is set to `/metrics_renders` and will be mounted on the docker container as `/renders` which where it'll save any output generated (despite why the log is saying, as it's intended to be displayed for action users):
https://github.com/lowlighter/metrics/blob/4141dd5a5bff259430398d328027afeeb74b03d8/source/app/action/index.mjs#L473-L474

The env vars are passed through an `--env-file` to be easier. 

Normally if the path is correctly mounted, renders *should* appear on the docker host. For example on GitHub Actions since the runner always mount `/metrics_renders:/renders` when running this app, results are accessed through the `/metrics_renders`, like in the workflow below:
https://github.com/lowlighter/metrics/blob/4141dd5a5bff259430398d328027afeeb74b03d8/.github/workflows/examples.yml#L1406-L1416

> I created a /renders, /metrics_renders, and /tmp folder at the root of my repo, but none of them contain the rendered svg

Not sure how you run the docker image, but if it's within a repository and you're running the docker image directly rather than using the GitHub Action, in addition to mounting the folder of your choice to `/renders`, you'll also need to manually commit and push your changes too to "save" the renders, like how it's done in the workflow above
